src/model/event.rs

```diff
@@ -404,7 +404,7 @@ impl CacheUpdate for GuildEmojisUpdateEvent {
 
     fn update(&mut self, cache: &mut Cache) -> Option<()> {
         cache.guilds.get_mut(&self.guild_id).map(|guild| {
-            guild.with_mut(|g| g.emojis.extend(self.emojis.clone()))
+            guild.with_mut(|g| g.emojis = self.emojis.clone())
```

Comment:
I don't like `(|foo| foo = bar)` syntax, can we make this:
```rust
guild.with_mut(|g| {
    g.emojis = ...;
});
```

Additionally, please use `HashMap::clone_from`.